libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/roadname/view/MapboxRoadNameView.kt

```diff
@@ -43,16 +43,24 @@ class MapboxRoadNameView : AppCompatTextView {
         defStyleAttr: Int
     ) : super(context, attrs, defStyleAttr)
 
+    /**
+     * The method is invoked automatically when the view is detached from the window and clears any
+     * relevant in memory cache.
+     */
+    override fun onDetachedFromWindow() {
+        shields.clear()
+        roadComponents.clear()
+        super.onDetachedFromWindow()
+    }
+
     /**
      * Invoke the method to render the view with road name.
```

Comment:
Should we add a comment here that we recommend to hide the view if the road components are empty? We are not doing it automatically (nor should we I guess).

Reply:
Yeah we should not automatically hide it. I think it's a good idea to add this recommendation.